x/dualstaking/keeper/grpc_query_delegator_providers.go

```diff
@@ -0,0 +1,59 @@
+package keeper
+
+import (
+	"context"
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/lavanet/lava/utils"
+	"github.com/lavanet/lava/x/dualstaking/types"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+func (k Keeper) DelegatorProviders(goCtx context.Context, req *types.QueryDelegatorProvidersRequest) (res *types.QueryDelegatorProvidersResponse, err error) {
+	if req == nil {
+		return nil, status.Error(codes.InvalidArgument, "invalid request")
+	}
+
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	var epoch uint64
+	if req.ShowPendingDelegators {
+		epoch, err = k.getNextEpoch(ctx)
+	} else {
+		epoch, _, err = k.epochstorageKeeper.GetEpochStartForBlock(ctx, uint64(ctx.BlockHeight()))
+	}
+	if err != nil {
+		return nil, err
+	}
+
+	providers, err := k.GetDelegatorProviders(ctx, req.Delegator, epoch)
+	if err != nil {
+		return nil, err
+	}
+
+	var delegations []types.Delegation
+	for _, provider := range providers {
+		indices := k.delegationFS.GetAllEntryIndicesWithPrefix(ctx, provider)
+		for _, ind := range indices {
+			delegator, providerDecode, chainID := types.DelegationKeyDecode(ind)
```

Comment:
`providerDecode` is not needed (must be same as `provider`), so replace with _

Reply:
done. see https://github.com/lavanet/lava/commit/d1e2d3c1afe278bf342ca8f975d03ca5326020c6